### PR TITLE
unhandicaps station drones

### PR DESCRIPTION
**Pull request by grungussuss**
## About The Pull Request
- makes area-based shy-ness for drones config
- reduces the cooldown on touching machines as a shy drone from 1 minute to 20 seconds
## Why It's Good For The Game
This was admin issues solved by codebase, no one plays drones anymore and those who do - complain that they're not fun because of the heavy restrictions imposed upon them.
If I remember correctly - the playtime restriction on our servers is 20 silicon hours? or 10 silicon hours? It's a config thing I can't check. This means it shouldn't be a problem with ban evaders and such.
Area restrictions are odd because drones can't get more air from atmos and they can't make power from SM, which is in their laws.
If a drone is messing with your SM then kill it, or ahelp if it keeps coming back?
For context:
```
"1. You may not involve yourself in the matters of another being, even if such matters conflict with Law Two or Law Three, unless the other being is another Drone.\n"+\
"2. You may not harm any being, regardless of intent or circumstance.\n"+\
"3. Your goals are to actively build, maintain, repair, improve, and provide power to the best of your abilities within the facility that housed your activation."
```
## Changelog
:cl: grungussuss
server: drone area restrictions are now config
balance: drones now take less time to become "un-shy" of something after it's been touched.
/:cl:

### CODE REVIEW

**Comment by carlarctg** [comment]
i'm 100% in favor of this, drone restrictions are absolutely ridiculous. you basically can't do anything that you are meant to be _able_ to do, and there are STILL plenty of ways to grief. 

add a lock of 100 silicon hours or something if necessary. but it's effectively unplayable rn, despite being a pretty cool concept.

do you want to know how many drone players i know of there are? one. in manuel. that's it. 

drones should either be removed or have restrictions loosened - this middleground is the worst of both worlds. right now the most impactful drones ever are is being a polymorph form.

**Comment by DrAmazing343** [comment]
As an admin, I've been saying this for a long while. We have the capacity to punish/enforce drone laws and I think we need to unhugbox them since it's so many silicon hours to unlock anyway- let them have fun!!

**Comment by DaydreamIQ** [comment]
You think you could also make it so drones are allowed to clean near people? It's infuriating that I'm not allowed to mop up bloodstains because john assistant went ssd too close to the medbay

**Comment by Sadboysuss** [comment]
> You think you could also make it so drones are allowed to clean near people? It's infuriating that I'm not allowed to mop up bloodstains because john assistant went ssd too close to the medbay

no

seperately ssd people shouldn't count for shy so if they do - make an issue report

**Comment by MrMelbert** [comment]
> This was admin issues solved by codebase

Note that admins refused to re-enable them unless the code base solved these problems

**Comment by Sadboysuss** [comment]
> Note that admins refused to re-enable them unless the code base solved these problems

is this still the case though or am I gonna get slapped

**Comment by DrAmazing343** [comment]
If the problem is brought up, I'll vote strongly in favor of keeping bots unhandicapped. If we see history repeating down the line, I may ask maintainers about a revert, but I think we can handle any trolls trying to ruin it for the rest of the folks

**Review by ZephyrTFA — CHANGES_REQUESTED** [review]
turn this into a config that defaults to enabled

**Comment by Sadboysuss** [comment]
> turn this into a config that defaults to enabled

[done](https://github.com/tgstation/tgstation/pull/87983/commits/24ceb21224b3f9227c6f9f394cad215f50c363e2)

**Review by scriptis — APPROVED** [review]

File: code/modules/mob/living/basic/drone/_drone.dm [hunk 1]
```diff
@@ -366,8 +369,8 @@
 		REMOVE_TRAIT(src, TRAIT_CAN_STRIP, DRONE_SHY_TRAIT) // To shy to touch someone elses hat
 		ADD_TRAIT(src, TRAIT_PACIFISM, DRONE_SHY_TRAIT)
 		LoadComponent(/datum/component/shy, mob_whitelist=not_shy_of, shy_range=3, message="Your laws prevent this action near %TARGET.", keyless_shy=FALSE, clientless_shy=TRUE, dead_shy=FALSE, dead_shy_immediate=TRUE, machine_whitelist=shy_machine_whitelist)
-		LoadComponent(/datum/component/shy_in_room, drone_bad_areas, "Touching anything in %ROOM could break your laws.")
-		LoadComponent(/datum/component/technoshy, 1 MINUTES, "%TARGET was touched by a being recently, using it could break your laws.")
+		init_shy_in_room_component(drone_bad_areas)
+		LoadComponent(/datum/component/technoshy, 20 SECONDS, "%TARGET was touched by a being recently, using it could break your laws.")
```
Comment (ZephyrTFA) [inline code/modules/mob/living/basic/drone/_drone.dm:373]: iffy about this one, but a minute is really harsh for ghost players who just want to do something

**Review by ZephyrTFA — APPROVED** [review]

**Pull request merged**